code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -0,0 +1,99 @@
+/obj/machinery/griddle
+	name = "griddle"
+	desc = "Because using pans is for pansies."
+	icon = 'icons/obj/machines/griddle.dmi'
+	icon_state = "griddle1_off"
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 5
+	layer = BELOW_OBJ_LAYER
+	circuit = /obj/item/circuitboard/machine/griddle
+	processing_flags = START_PROCESSING_MANUALLY
+	resistance_flags = FIRE_PROOF
+
+	///Things that are being griddled right now
+	var/list/griddled_objects = list()
+	///Looping sound for the grill
+	var/datum/looping_sound/grill/grill_loop
+	///Whether or not the machine is turned on right now
+	var/on = FALSE
+	///What variant of griddle is this?
+	var/variant = 1
+	///How many shit fits on the griddle?
+	var/max_items = 8
+
+/obj/machinery/griddle/Initialize()
+	. = ..()
+	grill_loop = new(list(src), FALSE)
+	variant = rand(1,3)
+
+/obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(griddled_objects.len >= max_items)
+		to_chat(user, "<span class='notice'>[src] can't fit more items!</span>")
+		return
```

Comment:
Is it possible to throw something onto the grill?

Reply:
not at the moment, not sure how I could best add it

Reply:
my inner idea guy has ideas, but they're not too important. I'm good with this.